core/lib/types/src/snapshots.rs

```diff
@@ -0,0 +1,80 @@
+use crate::commitment::L1BatchWithMetadata;
+use crate::{StorageKey, StorageValue, H256};
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use zksync_basic_types::{L1BatchNumber, MiniblockNumber};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AllSnapshots {
+    pub snapshots: Vec<SnapshotMetadata>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SnapshotMetadata {
+    pub l1_batch_number: L1BatchNumber,
+    pub generated_at: DateTime<Utc>,
+    pub factory_deps_filepath: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SnapshotHeader {
+    pub l1_batch_number: L1BatchNumber,
+    pub miniblock_number: MiniblockNumber,
+    pub storage_logs_chunks: Vec<SnapshotStorageLogsChunkMetadata>,
+    pub factory_deps_filepath: String,
+    pub last_l1_batch_with_metadata: L1BatchWithMetadata,
```

Comment:
`L1BatchWithMetadata` has a lot of unnecessary info, particularly in `metadata` and `factory_deps` fields. Do we really want to pass it in its entirety, or could it make sense to create a struct specifically for snapshot recovery? cc @RomanBrodetski

Reply:
I assumed that it doesn't cost us much to jut have those few additional fields while it lets us reuse lots of existing code and that it just wouldn't be worth it.

Reply:
The problem is not the cost, but rather compatibility that we'll probably need to care about (e.g., we want to change `L1BatchWithMetadata` and it now suddenly breaks snapshot recovery). Also, a cognitive overhead – if we use a dedicated data struct for snapshot generation, we *by design* know which data we use and why it's included (so it's easier to maintain), while with `L1BatchWithMetadata` it would be easier to end up with spaghetti code relying on `L1BatchWithMetadata` intrinsics.

Reply:
I will work on this in snapshots applier PR